### DISCUSSION TITLE
Support Internal Copy and Paste inside VS Code Dev Container

### DISCUSSION THREAD

**Discussion by cmcnicoll** [tconbeer/harlequin#645] [Q&A, Troubleshooting, Support] [answered]
### Current behavior 
- `ctrl+c` and `ctrl+x`:
	These commands throw the following error:
	```
	FileNotFoundError: [Errno 2] No such file or directory: 'clip.exe'
	```
- ctrl+v and ctrl+u:
	These commands throw the following error:
	```
	FileNotFoundError: [Errno 2] No such file or directory: 'powershell.exe'
	```
- `shift+insert` and right-click:
	These paste methods work correctly.

### Environment

I am using Windows 11 with Docker Desktop. I have tried several suggestions from the [troubleshooting guide](https://harlequin.sh/docs/troubleshooting/copying-and-pasting), but they did not resolve the issue.

### Questions

1. If I install xclip in my Dockerfile, can I use a [config file](https://harlequin.sh/docs/config-file) to override the current behavior?
1. Alternatively, is the solution to use Xming as discussed in [this thread](https://github.com/tconbeer/harlequin/discussions/431)?

Thanks!

**Comment by tconbeer** [comment, marked as answer]
I think something changed recently with pyperclip, or with the way VS code terminal (in WSL) sets some of its environment - I hit this issue myself yesterday. 

Right now we only catch pyperclip errors, and the other bubble up, including OSErrors. We should just catch all errors on copy/paste and fall back to using the internal clipboard only. 

(In other words this is a bug, and is a pretty easy fix)

**Reply by cmcnicoll** [reply]
I see you fixed this in #649. Thanks @tconbeer!
